cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationRunHttpHandler.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2014-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.gateway.handlers;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.security.AuditDetail;
+import io.cdap.cdap.common.security.AuditPolicy;
+import io.cdap.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import io.cdap.cdap.proto.operationrun.OperationRun;
+import io.cdap.cdap.spi.data.SortOrder;
+import io.cdap.http.HttpHandler;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import java.util.ArrayList;
+import java.util.List;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * The {@link HttpHandler} for handling REST calls to namespace endpoints.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/operations")
+public class OperationRunHttpHandler extends AbstractAppFabricHttpHandler {
+
+  private static final Gson GSON = new Gson();
+
+  @Inject
+  OperationRunHttpHandler() {
+  }
+
+  /**
+   * API to fetch all runs for an operation type in a namespace.
+   *
+   * @param namespaceId Namespace to fetch runs from
+   * @param operationType Type of operation
+   * @param pageToken the token identifier for the current page requested in a paginated
+   *     request
+   * @param pageSize the number of application details returned in a paginated request
+   * @param orderBy the sorting order in which results are returned, ASC for ascending, DESC for
+   *     descending
+   * @param stateFilter optional state to filter for
+   */
+  @GET
+  @Path("/{operation-type}/runs")
```

Comment:
these paths look different from the design, where the path is just `/namespaces/namespace/operations` without the type and 'runs' is not in the path.

In general, REST urls are `[collection]/[name]/[other collection]/[other name]`, so if there's anything after /operations, it would need to be the operation name. If we wanted the type in the path, it would be need to be something like `/namespaces/[namespace]/operationtypes/[type]/operations`